src/snowflake/snowpark/functions.py

```diff
@@ -327,6 +327,29 @@ def sql_expr(sql: str) -> Column:
     return Column._expr(sql)
 
 
+def reference(
+    object_type: str,
+    object_identifier: str,
+    scope: str = "CALL",
+    privileges: Optional[List[str]] = None,
+):
+    """
+    Returns a reference to an object (a table, view, or function). When you execute SQL actions on a
+    reference to an object, the actions are performed using the role of the user who created the
+    reference.
+
+    Example::
+        >>> df = session.create_dataframe([(1,)], schema=["A"])
+        >>> df.write.save_as_table("my_table", mode="overwrite", table_type="temporary")
+        >>> df.select(substr(reference("table", "my_table"), 1, 14).alias("identifier")).collect()
+        [Row(IDENTIFIER='ENT_REF_TABLE_')]
+    """
+    privileges = privileges or []
+    return builtin("system$reference")(
```

Comment:
Is there any existing function already using `system$` prefix? If not, maybe we need to be careful about how to add system functions? because if there were a function called `reference`, how do we add it?

Reply:
I agree with this. maybe we create a new file called `system_functions.py` to differentiate between regular and `system$` functions?

wdyt: @sfc-gh-yixie

Reply:
How about using the same module but renaming the function name to `system_reference`?